dockerfiles/theia/build.sh

```diff
@@ -15,3 +15,14 @@ build
 if ! skip_tests; then
   bash "${base_dir}"/e2e/build.sh "$@"
 fi
+
+echo "Extracting artifacts for the CDN"
+mkdir -p "${base_dir}/theia_artifacts"
+"${base_dir}"/extract-for-cdn.sh "$IMAGE_NAME" "${base_dir}/theia_artifacts"
+LABEL_CONTENT=$(cat "${base_dir}"/theia_artifacts/cdn.json || true 2>/dev/null)
+if [ "${LABEL_CONTENT}" != "" ]; then
+  BUILD_ARGS+="--label che-plugin.cdn.artifacts=$(echo ${LABEL_CONTENT} | sed 's/ //g') "
+  echo "Rebuilding with CDN label..."
```

Comment:
could we avoid to rebuild a second time ? or is it mandatory to build twice ?

Reply:
It seems it's mandatory to build twice, since the CDN.json file should have been created during the theia build in order to get its content and add it on the image label.
Unless we would know a way to change the label on an existing image, but I don't know any.

Reply:
it looks like it's more an extra build of the multi staged build then ?
(no need to use build.sh then)

Reply:
I mean, we could get rid of extract-for-cdn.sh as well

Reply:
if you look at the Dockerfile, there are two from instructions

From what I'm understanding, the json is generated on the first build (First from/Dockerfile-

so could we copy this json to the next build (second From of Dockerfile) add add it as a label ?

Reply:
The main problem I had was to add the label from *inside* the docker build.
You can use an environment variable (declared by `ENV`) in a `LABEL` instruction,
but I'm not aware of the ability to assign the result of a `bash` instruction (ie `cat cdn.json`) to such an environment variable from *inside* the docker build.

Reply:
yes, true it's not possible yet